src/odemis/driver/smarpod.py

```diff
@@ -870,3 +1117,568 @@ def Smarpod_GetDLLVersion(self, p_major, p_minor, p_update):
+                 rotary_speed=1, axes=None, **kwargs):
+        """
+        A driver for a SmarAct SA_MC Actuator.
+        This driver uses a DLL provided by SmarAct which connects via
+        USB or TCP/IP using a locator string.
+
+        name: (str)
+        role: (str)
+        locator: (str) Use "fake" for a simulator.
+            For a real device, MCS controllers with USB interface can be addressed with the
+            following locator syntax:
+                usb:id:<id>
+            where <id> is the first part of a USB devices serial number which
+            is printed on the MCS controller.
+            If the controller has a TCP/IP connection, use:
+                network:<ip>:<port>
+        ref_on_init: (bool) determines if the controller should automatically reference
+            on initialization
+        actuator_speed: (double) the default speed (in m/s) of the actuators
+        axes: dict str (axis name) -> dict (axis parameters)
+            axis parameters: {
+                range: [float, float], default is -1 -> 1
+                unit: (str) default will be set to 'm'
+            }
+        """
+        if not axes:
+            raise ValueError("Needs at least 1 axis.")
+
+        if locator != "fake":
+            self.core = SA_MCDLL()
+        else:
+            self.core = FakeMCDLL()
+
+        # Not to be mistaken with axes which is a simple public view
+        self._axis_map = {}  # axis name -> axis number used by controller
+        axes_def = {}  # axis name -> Axis object
+        self._locator = locator
+
+        for axis_name, axis_par in axes.items():
+            try:
+                axis_range = axis_par['range']
+            except KeyError:
+                logging.info("Axis %s has no range. Assuming (-1, 1)", axis_name)
+                axis_range = (-1, 1)
+
+            try:
+                axis_unit = axis_par['unit']
+            except KeyError:
+                logging.info("Axis %s has no unit. Assuming m", axis_name)
+                axis_unit = "m"
+
+            ad = model.Axis(canAbs=True, unit=axis_unit, range=axis_range)
+            axes_def[axis_name] = ad
+
+        # Connect to the device
+        self._id = c_uint32(SA_MCDLL.SA_MC_INVALID_HANDLE)
+
+        option_string = b"model " + str(SA_MCDLL.hwModel) + "\n" + \
+                        b"locator " + locator
+                        
+        options = c_char_p(option_string.encode("ascii"))
+
+        self.core.SA_MC_Open(byref(self._id), options)
+        logging.debug("Successfully connected to SA_MC Controller ID %d", self._id.value)
+        model.Actuator.__init__(self, name, role, axes=axes_def, **kwargs)
+
+        # Add metadata
+        # TODO: Fix getting software version with a supported function
+        self._swVersion = 0  # self.GetSwVersion()
+
+        self._metadata[model.MD_SW_VERSION] = self._swVersion
+        logging.debug("Using SA_MC library version %s", self._swVersion)
```

Comment:
Let's ask SmarAct to get this info. For now, just disable entirely the code, so that the default value ("unknown") is used.